examples/HelloWorldExampleDS/HelloWorldPublisher.cpp

```diff
@@ -204,12 +218,13 @@ void HelloWorldPublisher::run(
     thread.join();
 }
 
-bool HelloWorldPublisher::publish(bool waitForListener)
+bool HelloWorldPublisher::publish(
+        bool waitForListener)
 {
-    if (m_listener.firstConnected || !waitForListener || m_listener.n_matched>0)
+    if (m_listener.firstConnected || !waitForListener || m_listener.n_matched > 0)
```

Comment:
Here is where `firstConnected` is used, ensuring that there is at least one DataReader attached. But it is enough to check `m_listener.n_matched` (using the internal count that was implemented before). However, currently this variable is saving `total_count` (Total cumulative count the concerned endpoint discovered a match with another endpoint) and not `current_count` (Endpoints currently matched with the given endpoint). Thus, it could be if the endpoints are leaving at some point that a message is sent without any reader being matched.